exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/SqlConverter.java

```diff
@@ -364,4 +401,35 @@ private static SchemaPlus rootSchema(SchemaPlus schema) {
     }
   }
 
+  private static class DrillRexBuilder extends RexBuilder {
+    private DrillRexBuilder(RelDataTypeFactory typeFactory) {
+      super(typeFactory);
+    }
+
+    @Override
+    public RexNode ensureType(
+        RelDataType type,
+        RexNode node,
+        boolean matchNullability) {
+      RelDataType targetType = type;
+      if (matchNullability) {
+        targetType = matchNullability(type, node);
+      }
+      if (targetType.getSqlTypeName() == SqlTypeName.ANY) {
+        return node;
+      }
+      if (!node.getType().equals(targetType)) {
+        if(!targetType.isStruct()) {
+          final RelDataType anyType = TypeInferenceUtils.createCalciteTypeWithNullability(
```

Comment:
I thought Calcite might have mechanism to further catch the cases where implicit casting is not added in. 

Let me try if we can find any issue when there is no such a cast to ANY